h/accounts/__init__.py

```diff
@@ -34,8 +35,25 @@ def make_staff(username):
         raise NoSuchUserError
 
 
+def authenticated_user(request):
+    """Return the authenticated user or None.
+
+    :rtype: h.accounts.models.User or None
+
+    """
+    if not request.authenticated_userid:
+        return None
+
+    # pylint: disable=unpacking-non-sequence
+    username, _ = util.split_user(request.authenticated_userid)
```

Comment:
Done, and also added `request.auth_domain` to avoid duplication: <https://github.com/hypothesis/h/commit/c22efde798c59039cb3dd2313dee54a33abdaf0b>, <https://github.com/hypothesis/h/commit/f5d54b9627945186f362cb325766512f12ccb515>